client/src/partials/patient_registration/patient.js

```diff
@@ -86,6 +86,10 @@ angular.module('bhima.controllers')
     }
 
     function checkingExistPatient (file_number) {
```

Comment:
If I understand this code properly, if `file_number === 0`, `checkExistPatient()` returns a boolean, otherwise it returns a promise.  Wouldn't it be better to use `def.resolve(false)`?  That way you are return a consistent type.